### PR TITLE
Migrated AdaIN tutorial to keras 3

### PR DESCRIPTION
**Pull request by kharshith-k**
Please find my PR for migrating AdaIN tutorial from keras2 to keras3
[Colab Gist](https://colab.research.google.com/gist/kharshith-k/76a1c20d80aef2b918ef05251208158e/adain.ipynb#scrollTo=i1i5XWu61io8)

### CODE REVIEW

File: examples/generative/adain.py [hunk 1]
```diff
@@ -653,6 +669,7 @@ def on_epoch_end(self, epoch, logs=None):
         ax_content.set_title("Content Image")
         ax_reconstructed.imshow(reconstructed_image)
         ax_reconstructed.set_title("NST Image")
+    break
```
Comment (sachinprasadhs) [inline examples/generative/adain.py]: Why break here?

File: examples/generative/adain.py [hunk 2]
```diff
@@ -44,26 +45,24 @@
...
-seed for reproducibility. The global variables are hyperparameters
-which we can change as we like.
+We begin by importing the packages used throughout the example and
+defining a few global hyperparameters that control image size,
+batch size, and training duration.
 """
 
 import os
 import numpy as np
 import tensorflow as tf
```
Comment (sachinprasadhs) [inline examples/generative/adain.py]: Are we using tf anywhere? if not, we can remove them

**Review by sachinprasadhs — COMMENTED** [review]
Thank, just small comments

File: examples/generative/adain.py (hunk 1 above)
Comment (kharshith-k) [reply examples/generative/adain.py]: It was used to infer on a single batch, however I refactored the code since it looked misleading

File: examples/generative/adain.py [hunk 3]
```diff
@@ -109,110 +107,161 @@
...
-    tf.data.Dataset.from_tensor_slices(val_style)
-    .map(decode_and_resize, num_parallel_calls=AUTOTUNE)
-    .repeat()
-)
-val_content_ds = (
-    tfds.load("voc", split="validation").map(extract_image_from_voc).repeat()
-)
+    train = paths[:train_end]
+    val = paths[train_end:val_end]
+    test = paths[val_end:]
```
Comment (maitry63) [inline examples/generative/adain.py:200]: Here, it might be better to shuffle the image paths before splitting them. 
Now, the split depends on the file order.

File: examples/generative/adain.py [hunk 4]
```diff
@@ -109,110 +107,161 @@
...
+train_ds = StyleContentPyDataset(
+    train_style,
+    train_content,
+    batch_size=BATCH_SIZE,
+    image_size=IMAGE_SIZE,
+    shuffle=True,
+)
+val_ds = StyleContentPyDataset(
+    val_style, val_content, batch_size=BATCH_SIZE, image_size=IMAGE_SIZE, shuffle=True
+)
```
Comment (maitry63) [inline examples/generative/adain.py:262]: @kharshith-k, you could set `shuffle=False` for the val_ds. This makes the validation results more consistent across epochs.

File: examples/generative/adain.py [hunk 5]
```diff
@@ -44,26 +45,23 @@
...
 import matplotlib.pyplot as plt
-import tensorflow_datasets as tfds
-from tensorflow.keras import layers
+from keras import layers
 
 # Defining the global variables.
 IMAGE_SIZE = (224, 224)
 BATCH_SIZE = 64
 # Training for single epoch for time constraint.
 # Please use atleast 30 epochs to see good results.
```
Comment (laxmareddyp) [inline examples/generative/adain.py]: nit:  Training for 30 epochs. Reduce for faster iteration.
 Please use at least 30 epochs to see good results.

File: examples/generative/adain.py
Comment (laxmareddyp) [inline examples/generative/adain.py]: The Introduction links to https://www.tensorflow.org/tutorials/generative/style_transfer — a TensorFlow-specific tutorial. This is now a Keras 3 backend-agnostic example; use a neutral reference instead.

ex: 
https://en.wikipedia.org/wiki/Neural_style_transfer

File: examples/generative/adain.py [hunk 6]
```diff
@@ -599,7 +615,8 @@ def on_epoch_end(self, epoch, logs=None):
 the loss function and then train it.
 
 *Note*: We train the model for a single epoch for time constraints,
```
Comment (laxmareddyp) [inline examples/generative/adain.py]: nit : Note*: We train the model for 30 epochs here, but you can reduce this for faster iteration.

File: examples/generative/adain.py [hunk 7]
```diff
@@ -2,9 +2,10 @@
 Title: Neural Style Transfer with AdaIN
 Author: [Aritra Roy Gosthipaty](https://twitter.com/arig23498), [Ritwik Raha](https://twitter.com/ritwik_raha)
 Date created: 2021/11/08
-Last modified: 2021/11/08
+Last modified: 2026/04/13
 Description: Neural Style Transfer with Adaptive Instance Normalization.
 Accelerator: GPU
+Converted to Keras 3 by: [Harshith K](https://github.com/kharshith-k/)
```
Comment (laxmareddyp) [inline examples/generative/adain.py]: Converted to Keras 3 by: it is a non-standard metadata field. Most keras-io tutorials don't have this. Consider removing it and crediting in the PR description instead.

File: examples/generative/adain.py
Comment (laxmareddyp) [inline examples/generative/adain.py]: The docstring lists VGG layers used for style loss as block1_conv1 through block1_conv4, but the code at line 482 uses block1_conv1, block2_conv1, block3_conv1, block4_conv1. it's pre existing but worth fixing while migrating.

Suggession: 

- `block1_conv1`
- `block2_conv1`
- `block3_conv1`
- `block4_conv1`

File: examples/generative/adain.py [hunk 8]
```diff
@@ -635,24 +655,24 @@ def on_epoch_end(self, epoch, logs=None):
...
-    ):
-        ax_style, ax_content, ax_reconstructed = axis
-        ax_style.imshow(style_image)
-        ax_style.set_title("Style Image")
-        ax_content.imshow(content_image)
-        ax_content.set_title("Content Image")
-        ax_reconstructed.imshow(reconstructed_image)
-        ax_reconstructed.set_title("NST Image")
+for axis, style_image, content_image, reconstructed_image in zip(
+    axes, style[0:10], content[0:10], reconstructed_image[0:10]
```
Comment (laxmareddyp) [inline examples/generative/adain.py]: nit: Variable name reconstructed_image is used both for the full batch tensor line 662 and as the loop iteration variable. This shadows the outer variable. Rename the loop variable to e.g. recon_image.

File: examples/generative/adain.py [hunk 9]
```diff
@@ -223,7 +277,7 @@ def extract_image_from_voc(element):
 from our dataset.
 """
 
-style, content = next(iter(train_ds))
+((style, content),) = next(iter(train_ds))
```
Comment (laxmareddyp) [inline examples/generative/adain.py]: nit: Uses next(iter(train_ds)) to get one batch, but PyDataset supports direct indexing. Consider train_ds[0] for consistency with line 658 where test_ds[0] is already used.

**Review by laxmareddyp — CHANGES_REQUESTED** [review]
Thanks for the PR..! Left few comments, address it.

**Review by laxmareddyp — APPROVED** [review]
Thanks..!LGTM.
